### PR TITLE
Feat/samlv2

### LINKED ISSUE #366: Have SSO on the Community edition
**Having SSO starting the community edition**

SSO is now a standard security practice, and should **not** be restricted to paid offers, especially that it will pull the MFA in modern configurations. We shall move this to the community edition as part of our social commitment, and simply because it's the right thing to do, especially for a cybersecurity solution.

### PR DESCRIPTION
**Pull request by nas-tabchiche**
- [x] Add SSO/SAML connection
- [x] configure IdP from the front-end
- [x] Expose ACS endpoints for docker/production

Fix #366

### CODE REVIEW

**Comment by AndrzejRPiotrowski** [comment]
is this working ?

**Comment by eric-intuitem** [comment]
@AndrzejRPiotrowski  this is a work in progress, stay tuned, should be ready by end of week.

File: backend/iam/sso/serializers.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        model = SSOSettings
+        exclude = ["value"]
+
+    def update(self, instance, validated_data):
+        settings_object = GlobalSettings.objects.get(name=GlobalSettings.Names.SSO)
+        settings_object.value = validated_data
+        settings_object.save()
+        return instance
+
+    # def build_saml_settings(self, validated_data):
```
Comment (Mohamed-Hacene) [inline backend/iam/sso/serializers.py]: remove commented block

File: backend/iam/views.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ class LogoutView(views.APIView):
     def post(self, request) -> Response:
         try:
             logger.info("logout request", user=request.user)
+            print("logout request", request.user)
```
Comment (Mohamed-Hacene) [inline backend/iam/views.py]: remove debug print

File: frontend/src/lib/components/Modals/CreateModal.svelte [hunk 3]
```diff
@@ -42,6 +42,6 @@
 				<i class="fa-solid fa-xmark" />
 			</div>
 		</div>
-		<ModelForm {form} {suggestions} {parent} {model} {closeModal} action="?/{formAction}" {debug} />
+		<ModelForm {form} {suggestions} {parent} {model} {closeModal} action="?/{formAction}" debug />
```
Comment (Mohamed-Hacene) [inline frontend/src/lib/components/Modals/CreateModal.svelte]: remove debug

File: frontend/src/routes/(app)/[model=urlmodel]/[id=uuid]/edit/+page.svelte [hunk 4]
```diff
@@ -15,4 +15,5 @@
 	foreignKeys={data.foreignKeys}
 	model={data.model}
 	origin="edit"
+	debug
```
Comment (Mohamed-Hacene) [inline frontend/src/routes/(app)/[model=urlmodel]/[id=uuid]/edit/+page.svelte]: remove debug

**Review by Mohamed-Hacene — COMMENTED** [review]
Seems ok for cleaning

File: backend/iam/models.py [hunk 5]
```diff
@@ -243,7 +243,7 @@ def _create_user(self, email, password, mailing=True, **extra_fields):
             folder=_get_root_folder(),
         )
         user.user_groups.set(extra_fields.get("user_groups", []))
-        user.password = make_password(password if password else str(uuid.uuid4()))
```
Comment (ab-smith) [inline backend/iam/models.py]: we might need to challenge this

**Review by eric-intuitem — APPROVED** [review]
LGTM

**Review by ab-smith — APPROVED** [review]

**Pull request merged**